### PR TITLE
feat: Improve Settings order

### PR DESCRIPTION
**Pull request by validcube**
Improve Settings

image

### CODE REVIEW

File: app/src/main/res/values/strings.xml [hunk 1]
```diff
@@ -49,19 +52,18 @@
...
     <string name="general">General</string>
-    <string name="general_description">General settings</string>
-    <string name="advanced">Advanced</string>
-    <string name="advanced_description">Advanced settings</string>
+    <string name="general_description">Theme, dynamic color</string>
     <string name="updates">Updates</string>
-    <string name="updates_description">Updates for ReVanced Manager</string>
-    <string name="opensource_licenses">Open source licenses</string>
-    <string name="opensource_licenses_description">View all the libraries used to make this application</string>
+    <string name="updates_description">Check for Update, changelogs</string>
```
Comment (oSumAtrIX) [inline app/src/main/res/values/strings.xml:57]:
```suggestion
    <string name="updates_description">Check for updates and view changelogs</string>
```

**Comment by Axelen123** [comment]
I think the "general" screen should be renamed. It only contains appearance settings

**Comment by oSumAtrIX** [comment]
How about "Visual" or "Design"?

File: app/src/main/res/values/strings.xml [hunk 2]
```diff
@@ -49,19 +52,18 @@
...
     <string name="downloads">Downloads</string>
-    <string name="downloads_description">Manage downloaded content</string>
+    <string name="downloads_description">Split APK, downloaded application</string>
```
Comment (Axelen123) [inline app/src/main/res/values/strings.xml:59]: This text is more accurate with the downloader plugin system branch.
```suggestion
    <string name="downloads_description">Downloader plugins and downloaded apps</string>
```

Comment (oSumAtrIX) [reply app/src/main/res/values/strings.xml:59]: How about "Downloaders and apps"?

Comment (oSumAtrIX) [reply app/src/main/res/values/strings.xml:59]: How about:

```suggestion
    <string name="downloads_description">Downloaders and apps</string>
```

File: app/src/main/java/app/revanced/manager/ui/screen/settings/AdvancedSettingsScreen.kt [hunk 3]
```diff
@@ -138,16 +137,46 @@ fun AdvancedSettingsScreen(
...
+                            ClipData.newPlainText("Device Information", deviceContent)
+                        )
+
+                        /* TODO: Investigate & Testing
+                            https://developer.android.com/develop/ui/views/touch-and-input/copy-paste#duplicate-notifications
+                            Android would like us to show Toast for API 32 or lower.
+
+                            Some custom skin don't actually support this.
+                            if (Build.VERSION.SDK_INT <= Build.VERSION_CODES.S_V2)
+                        */
```
Comment (validcube) [inline app/src/main/java/app/revanced/manager/ui/screen/settings/AdvancedSettingsScreen.kt:172]:
```suggestion
```

File: app/src/main/java/app/revanced/manager/ui/screen/settings/AdvancedSettingsScreen.kt [hunk 4]
```diff
@@ -138,16 +137,39 @@ fun AdvancedSettingsScreen(
...
+            SettingsListItem(
+                modifier = Modifier.combinedClickable(
+                    onLongClickLabel = stringResource(R.string.copy_to_clipboard),
+                    onLongClick = {
+                        haptics.performHapticFeedback(HapticFeedbackType.LongPress)
+                        clipboard.setPrimaryClip(
+                            ClipData.newPlainText("Device Information", deviceContent)
+                        )
+
+                        Toast.makeText(context, R.string.toast_copied_to_clipboard, Toast.LENGTH_SHORT).show()
```
Comment (Axelen123) [inline app/src/main/java/app/revanced/manager/ui/screen/settings/AdvancedSettingsScreen.kt:166]: Use the `Context.toast` extension function from `Utils.kt`.

File: app/src/main/java/app/revanced/manager/ui/screen/settings/AdvancedSettingsScreen.kt [hunk 5]
```diff
@@ -138,16 +137,39 @@ fun AdvancedSettingsScreen(
...
+                    }
+                ) {
+
+                },
```
Comment (Axelen123) [inline app/src/main/java/app/revanced/manager/ui/screen/settings/AdvancedSettingsScreen.kt:170]: I think doing `onClick = {}` would be more readable

File: app/src/main/java/app/revanced/manager/ui/screen/settings/AdvancedSettingsScreen.kt [hunk 6]
```diff
@@ -82,15 +90,6 @@ fun AdvancedSettingsScreen(
...
             )
 
-            val exportDebugLogsLauncher =
-                rememberLauncherForActivityResult(ActivityResultContracts.CreateDocument("text/plain")) {
-                    it?.let(vm::exportDebugLogs)
-                }
-            SettingsListItem(
-                headlineContent = stringResource(R.string.debug_logs_export),
-                modifier = Modifier.clickable { exportDebugLogsLauncher.launch(vm.debugLogFileName) }
-            )
```
Comment (Axelen123) [inline app/src/main/java/app/revanced/manager/ui/screen/settings/AdvancedSettingsScreen.kt:92]: Why was this removed?

File: app/src/main/java/app/revanced/manager/ui/screen/settings/AdvancedSettingsScreen.kt [hunk 7]
```diff
@@ -138,16 +137,39 @@ fun AdvancedSettingsScreen(
...
-                    **Version**: ${BuildConfig.VERSION_NAME} (${BuildConfig.VERSION_CODE})
-                    **Build type**: ${BuildConfig.BUILD_TYPE}
-                    **Model**: ${Build.MODEL}
-                    **Android version**: ${Build.VERSION.RELEASE} (${Build.VERSION.SDK_INT})
-                    **Supported Archs**: ${Build.SUPPORTED_ABIS.joinToString(", ")}
-                    **Memory limit**: $memoryLimit
+                headlineContent = stringResource(R.string.debug_logs_export),
+                modifier = Modifier.clickable { exportDebugLogsLauncher.launch(vm.debugLogFileName) }
+            )
+            val clipboard = context.getSystemService(CLIPBOARD_SERVICE) as ClipboardManager
```
Comment (Axelen123) [inline app/src/main/java/app/revanced/manager/ui/screen/settings/AdvancedSettingsScreen.kt:148]: Use the Kotlin version of this function (like at the top of the file) and wrap this inside a `remember` block.

**Review by Axelen123 — CHANGES_REQUESTED** [review]

File: app/src/main/java/app/revanced/manager/ui/screen/settings/AdvancedSettingsScreen.kt (hunk 7 above)
Comment (validcube) [reply app/src/main/java/app/revanced/manager/ui/screen/settings/AdvancedSettingsScreen.kt]: Done

File: app/src/main/java/app/revanced/manager/ui/screen/settings/AdvancedSettingsScreen.kt (hunk 6 above)
Comment (validcube) [reply app/src/main/java/app/revanced/manager/ui/screen/settings/AdvancedSettingsScreen.kt:92]: It's not removed, it's relocated

File: app/src/main/java/app/revanced/manager/ui/screen/settings/AdvancedSettingsScreen.kt (hunk 5 above)
Comment (validcube) [reply app/src/main/java/app/revanced/manager/ui/screen/settings/AdvancedSettingsScreen.kt]: Done

File: app/src/main/java/app/revanced/manager/ui/screen/settings/AdvancedSettingsScreen.kt (hunk 4 above)
Comment (validcube) [reply app/src/main/java/app/revanced/manager/ui/screen/settings/AdvancedSettingsScreen.kt]: Done

File: app/src/main/java/app/revanced/manager/ui/screen/settings/AdvancedSettingsScreen.kt (hunk 6 above)
Comment (Axelen123) [reply app/src/main/java/app/revanced/manager/ui/screen/settings/AdvancedSettingsScreen.kt:92]: Oh, it looked like it was removed in the diff but now I see it is there.

File: app/src/main/java/app/revanced/manager/ui/screen/settings/AdvancedSettingsScreen.kt [hunk 8]
```diff
@@ -138,16 +138,38 @@ fun AdvancedSettingsScreen(
...
-                    **Version**: ${BuildConfig.VERSION_NAME} (${BuildConfig.VERSION_CODE})
-                    **Build type**: ${BuildConfig.BUILD_TYPE}
-                    **Model**: ${Build.MODEL}
-                    **Android version**: ${Build.VERSION.RELEASE} (${Build.VERSION.SDK_INT})
-                    **Supported Archs**: ${Build.SUPPORTED_ABIS.joinToString(", ")}
-                    **Memory limit**: $memoryLimit
+                headlineContent = stringResource(R.string.debug_logs_export),
+                modifier = Modifier.clickable { exportDebugLogsLauncher.launch(vm.debugLogFileName) }
+            )
+            val clipboard = remember { context.getSystemService(CLIPBOARD_SERVICE) as ClipboardManager }
```
Comment (Axelen123) [inline app/src/main/java/app/revanced/manager/ui/screen/settings/AdvancedSettingsScreen.kt:149]: Use the extension with this signature: `context.getSystemService<ClipboardManager>()!!` for the sake of consistency.

**Review by Axelen123 — CHANGES_REQUESTED** [review]

**Comment by oSumAtrIX** [comment]
Please update the attached image in OP as well

File: app/src/main/java/app/revanced/manager/ui/screen/settings/AdvancedSettingsScreen.kt (hunk 8 above)
Comment (validcube) [inline app/src/main/java/app/revanced/manager/ui/screen/settings/AdvancedSettingsScreen.kt:149]:
```suggestion
            val clipboard = remember { context.getSystemService<ClipboardManager>() }
```

File: app/src/main/java/app/revanced/manager/ui/screen/settings/AdvancedSettingsScreen.kt [hunk 9]
```diff
@@ -138,16 +138,38 @@ fun AdvancedSettingsScreen(
...
-                    **Version**: ${BuildConfig.VERSION_NAME} (${BuildConfig.VERSION_CODE})
-                    **Build type**: ${BuildConfig.BUILD_TYPE}
-                    **Model**: ${Build.MODEL}
-                    **Android version**: ${Build.VERSION.RELEASE} (${Build.VERSION.SDK_INT})
-                    **Supported Archs**: ${Build.SUPPORTED_ABIS.joinToString(", ")}
-                    **Memory limit**: $memoryLimit
+                headlineContent = stringResource(R.string.debug_logs_export),
+                modifier = Modifier.clickable { exportDebugLogsLauncher.launch(vm.debugLogFileName) }
+            )
+            val clipboard = remember { context.getSystemService<ClipboardManager>() }
```
Comment (validcube) [inline app/src/main/java/app/revanced/manager/ui/screen/settings/AdvancedSettingsScreen.kt:149]:
```suggestion
            val clipboard = remember { context.getSystemService<ClipboardManager>()!! }
```

File: app/src/main/java/app/revanced/manager/ui/screen/settings/AdvancedSettingsScreen.kt (hunk 8 above)
Comment (validcube) [reply app/src/main/java/app/revanced/manager/ui/screen/settings/AdvancedSettingsScreen.kt]: Done

**Comment by validcube** [comment]
This PR also additionally allow making a listview without description for self-explanatory label.

File: app/src/main/res/values/strings.xml [hunk 10]
```diff
@@ -49,19 +52,18 @@
...
-    <string name="updates_description">Updates for ReVanced Manager</string>
-    <string name="opensource_licenses">Open source licenses</string>
-    <string name="opensource_licenses_description">View all the libraries used to make this application</string>
+    <string name="updates_description">Check for updates and view changelogs</string>
     <string name="downloads">Downloads</string>
-    <string name="downloads_description">Manage downloaded content</string>
+    <string name="downloads_description">Downloader plugins and downloaded apps</string>
     <string name="import_export">Import &amp; export</string>
-    <string name="import_export_description">Import and export settings</string>
+    <string name="import_export_description">Keystore, patch option &amp; selection</string>
```
Comment (validcube) [inline app/src/main/res/values/strings.xml:61]:
```suggestion
    <string name="import_export_description">Keystore, patch option and selection</string>
```

Consistency sake, ampersand should not be use.

File: app/src/main/java/app/revanced/manager/ui/screen/settings/AdvancedSettingsScreen.kt [hunk 11]
```diff
@@ -1,10 +1,15 @@
 package app.revanced.manager.ui.screen.settings
 
 import android.app.ActivityManager
+import android.content.ClipData
+import android.content.ClipboardManager
+import android.content.Context.CLIPBOARD_SERVICE
```
Comment (Axelen123) [inline app/src/main/java/app/revanced/manager/ui/screen/settings/AdvancedSettingsScreen.kt:6]: This is unused

**Review by Axelen123 — CHANGES_REQUESTED** [review]

File: app/src/main/java/app/revanced/manager/ui/screen/settings/AdvancedSettingsScreen.kt (hunk 11 above)
Comment (validcube) [reply app/src/main/java/app/revanced/manager/ui/screen/settings/AdvancedSettingsScreen.kt]: fixed

File: app/src/main/res/values/strings.xml [hunk 12]
```diff
@@ -49,19 +52,18 @@
...
-    <string name="updates_description">Updates for ReVanced Manager</string>
-    <string name="opensource_licenses">Open source licenses</string>
-    <string name="opensource_licenses_description">View all the libraries used to make this application</string>
+    <string name="updates_description">Check for updates and view changelogs</string>
     <string name="downloads">Downloads</string>
-    <string name="downloads_description">Manage downloaded content</string>
+    <string name="downloads_description">Downloader plugins and downloaded apps</string>
     <string name="import_export">Import &amp; export</string>
-    <string name="import_export_description">Import and export settings</string>
+    <string name="import_export_description">Keystore, patch option and selection</string>
```
Comment (Axelen123) [inline app/src/main/res/values/strings.xml:61]:
```suggestion
    <string name="import_export_description">Keystore, patch options and selection</string>
```

File: app/src/main/res/values/strings.xml [hunk 13]
```diff
@@ -49,19 +52,18 @@
...
+    <string name="advanced">Advanced</string>
+    <string name="advanced_description">API URL, memory limit, debug</string>
```
Comment (Axelen123) [inline app/src/main/res/values/strings.xml:63]:
```suggestion
    <string name="advanced_description">API URL, memory limit, debugging</string>
```

**Review by Axelen123 — CHANGES_REQUESTED** [review]

**Review by Axelen123 — APPROVED** [review]
